packages/cascara/src/modules/DataCheckbox/DataCheckbox.js

```diff
@@ -31,26 +32,37 @@ const DataCheckbox = ({
   const { isEditing, formMethods } = useContext(ModuleContext);
   const checkbox = useCheckboxState({ state: Boolean(value) });
 
+  const conditionalLabelProps = getConditionalLabelProps(label, isLabeled);
+
   const renderEditing = (
-    <label htmlFor={label}>
+    <>
+      {label && isLabeled && (
+        <label htmlFor={attribute || label}>
+          <span className={styles.LabelText}>{label || attribute}</span>
+        </label>
+      )}
       <Checkbox
+        {...conditionalLabelProps}
         {...rest}
         {...checkbox}
         className={styles.Input}
-        id={label}
         name={attribute || label}
         ref={formMethods?.register}
       />
       {label && isLabeled && <span className={styles.LabelText}>{label}</span>}
-    </label>
+    </>
```

Comment:
I just happened to catch this luckily. For a checkbox, the <label> tag is supposed to wrap the checkbox input itself. It looks like that is being changed here.

Reply:
Thanks for catching this! The input component must be wrapped by the label, IDK how I missed this - Per our convo in slack, a new `LabeledModule` was created to abstract this logic away from modules.

Modules were cleanned up and code is much easier to follow 👍 
Please let me know if you still see improvement opportunities, this has to be in very good shape for the release.